tests/test_new_command.rs

```diff
@@ -129,6 +129,69 @@ fn test_new_insert_setup() {
     "###);
 }
 
+#[test]
+fn test_new_insert_after() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    setup_before_insertion(&test_env, &repo_path);
+
+    let stdout =
+        test_env.jj_cmd_success(&repo_path, &["new", "--insert-after", "-m", "G", "B", "D"]);
+    insta::assert_snapshot!(stdout, @r###"
+    Rebased 2 descendant commits
+    Working copy now at: aa74099a6658 G
+    "###);
+    insta::assert_snapshot!(get_short_log_output(&test_env, &repo_path), @r###"
+    o C
+    | o   F
+    | |\  
+    |/ /  
+    @ |   G
+    |\ \  
+    | | o E
+    o | | D
+    | |/  
+    |/|   
+    | o B
+    | o A
+    |/  
+    o root
+    "###);
+}
+
+#[test]
+fn test_new_insert_after_children() {
```

Comment:
You should have a test of `jj new --after D`. I suspect your code will replace both parents of `G` with the new commit as opposed to just `D`.

Reply:
It looks like it works fine. I've added this command to insert node H, and I get (also included in the automated tests):

```
    o C
    | o   F
    | |\  
    |/ /  
    o |   G
    |\ \  
    @ | | H
    | | o E
    o | | D
    | |/  
    |/|   
    | o B
    | o A
    |/  
    o root
```

Reply:
Indeed, it seems to work great. Nice! Thanks for adding the test.